src/stream.rs

```diff
@@ -11,27 +11,29 @@ use chunked_transfer::Decoder as ChunkDecoder;
 #[cfg(feature = "socks-proxy")]
 use socks::{TargetAddr, ToTargetAddr};
 
+#[cfg(not(feature = "native-tls"))]
+use crate::default_tls_config;
 use crate::proxy::Proxy;
 use crate::{error::Error, proxy::Proto};
 
 use crate::error::ErrorKind;
 use crate::unit::Unit;
 
 /// Trait for things implementing [std::io::Read] + [std::io::Write]. Used in [TlsConnector].
-pub trait ReadWrite: Read + Write + Send + 'static {
+pub trait ReadWrite: Read + Write + Send + Sync + 'static {
```

Comment:
This is a breaking change in our public API. Would required a major version bump.

Reply:
This was required to get the .src() call working in case of an error.
std::io::Error requires 'Sync'.

Perhaps there's a different solution to this?